handler/src/main/java/io/netty/handler/ssl/ApplicationProtocolNegotiationHandler.java

```diff
@@ -145,9 +145,19 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             }
         }
 
+        if (evt instanceof ChannelInputShutdownEvent) {
```

Comment:
It is a good idea to fire messages in case of channel shutdown? Let's say if the handshake fails for some reason, the user should not be expecting any message to reach his handler. Maybe we should drain the message entirely in case of failure.

Reply:
@normanmaurer WDYT?

Reply:
if the handshake fails the `SslHandler` should not forward any so no special handling should be done here